### ISSUE TITLE
Open source license

### ISSUE THREAD

**Issue by JBontes** [CalebLBaker/b-tree#2]
I would like to use a btree implementation in my source, but I cannot because your code does not have a license.
Can you pls add a suitable OSS license?

Something like MIT or Apache?

**Comment by CalebLBaker** [comment]
I will add the MIT license when I get a chance, but I also want to make sure you're aware that this was a project I did for school that I've never really given any thought to again after finishing the class it was for. It's unmaintained and has no suite of tests associated with it to verify correct behavior (I got an A on the assignment so clearly my professor didn't spot any bugs, but that's no excuse for not having unit tests).

I would recommend using a more industry grade b-tree implementation unless there is a specific reason that my implementation suites your purposes better.

Here is a library that is actively maintained and tested which contains a good b-tree implementation:
https://github.com/abseil/abseil-cpp

If you don't want to pull in a large library like that just for it's b-tree data structures, here is another library which, while no longer actively maintained, was done more professionally and is better tested than mine and which just contains a b-tree implementation and not a million other things:
https://github.com/diegocaro/cpp-btree
